src/firefly/js/visualize/XYPlotCntlr.js

```diff
@@ -154,103 +170,56 @@ export function reducer(state={}, action={}) {
+            if (state[tblId].xyPlotParams === xyPlotParams) {
+                const decimatedUnzoomedNext = isUndefined(decimatedUnzoomed) ? state[tblId].decimatedUnzoomed : decimatedUnzoomed;
+                const newParams = xyPlotParams.selection ?  updateSet(xyPlotParams, 'selection', undefined) : xyPlotParams;
+                return updateSet(state, tblId,
+                    {isPlotDataReady, decimatedUnzoomed: decimatedUnzoomedNext, xyPlotData, xyPlotParams: newParams});
+            }
+            return state;
         }
         case (SET_SELECTION) :
         {
             const {tblId, selection} = action.payload;
-            return update(state, {[tblId] : {xyPlotParams: {selection: {$set: selection}}}});
+            return updateSet(state, [tblId,'xyPlotParams','selection'], selection);
         }
         case (SET_ZOOM) :
         {
             const {tblId, selection} = action.payload;
-            const newState = update(state,
-                {[tblId] :
-                    {xyPlotParams:
-                        {
-                            selection: {$set: undefined},
-                            zoom: {$set: selection}
-                         }
-                    }});
-            if  (get(newState[tblId], 'xyPlotData.decimateKey')) {
-                const nextPlotParams = newState[tblId].xyPlotParams;
-                newState[tblId] = {plotDataReady: false, decimatedUnzoomed: state[tblId].decimatedUnzoomed};
-                const request = findTblById(tblId).request;
-                action.sideEffect((dispatch) => fetchPlotData(dispatch, request, nextPlotParams));
-            }
+            const newState = updateSet(state, [tblId,'xyPlotParams','zoom'], selection);
+            Reflect.deleteProperty(newState[tblId].xyPlotParams, 'selection');
             return newState;
         }
         case (RESET_ZOOM) :
         {
             const tblId = action.payload.tblId;
-            const newState = update(state,
-                {[tblId] :
-                    {xyPlotParams:
-                        {
-                            selection: {$set: undefined},
-                            zoom: {$set: undefined}
-                        }
-                    }});
-            if  (state[tblId].decimatedUnzoomed || isUndefined(state[tblId].decimatedUnzoomed)) {
-                const nextPlotParams = newState[tblId].xyPlotParams;
-                newState[tblId] = {plotDataReady: false};
-                const request = findTblById(tblId).request;
-                action.sideEffect((dispatch) => fetchPlotData(dispatch, request, nextPlotParams));
-            }
-            return newState;
-
+            const newParams = Object.assign({}, state[tblId].xyPlotParams);
+            Reflect.deleteProperty(newParams, 'selection');
+            Reflect.deleteProperty(newParams, 'zoom');
+            return updateSet(state, [tblId,'xyPlotParams'], newParams);
```

Comment:
lodash omit is a good option for this.
const newParams = omit(state[tblId].xyPlotParams, ['selection', 'zoom']);